File: ports/1.21.1-full/src/main/java/mods/eln/entity/ReplicatorRender.java
package mods.eln.entity;

import net.minecraft.client.model.ModelBase;
import net.minecraft.client.renderer.entity.MobRenderer;
import net.minecraft.client.renderer.entity.EntityRenderDispatcher;
import net.minecraft.world.entity.Entity;
import net.minecraft.resources.ResourceLocation;

public class ReplicatorRender extends MobRenderer {

    private static final ResourceLocation res = new ResourceLocation("eln:textures/entity/replicator.png");

    public ReplicatorRender(EntityRenderDispatcher rendermanagerIn, ModelBase modelbaseIn, float shadowsizeIn) {
        super(rendermanagerIn, modelbaseIn, shadowsizeIn);
    }

    @Override
    protected ResourceLocation getEntityTexture(Entity entity) {
        return res;
    }
}

File: src/main/kotlin/mods/eln/entity/ReplicatorRender.kt
package mods.eln.entity

import net.minecraft.client.model.SilverfishModel
import net.minecraft.client.renderer.entity.EntityRendererProvider
import net.minecraft.client.renderer.entity.MobRenderer
import net.minecraft.resources.ResourceLocation

/** Upstream draws the replicator with vanilla's silverfish model and its own texture. */
class ReplicatorRender(context: EntityRendererProvider.Context, model: SilverfishModel<ReplicatorEntity>, shadowSize: Float) :
    MobRenderer<ReplicatorEntity, SilverfishModel<ReplicatorEntity>>(context, model, shadowSize) {

    override fun getTextureLocation(entity: ReplicatorEntity): ResourceLocation = RES

    companion object {
        private val RES = ResourceLocation.fromNamespaceAndPath("eln", "textures/entity/replicator.png")
    }
}
